packages/thirdweb/src/extensions/marketplace/direct-listings/write/buyFromListing.ts

```diff
@@ -71,3 +71,19 @@
     },
   });
 }
+
+/**
+ * Checks if the `buyFromListing` method is supported by the given contract.
+ * @param availableSelectors An array of 4byte function selectors of the contract. You can get this in various ways, such as using "whatsabi" or if you have the ABI of the contract available you can use it to generate the selectors.
+ * @returns A boolean indicating if the `buyFromListing` method is supported.
+ * @extension MARKETPLACE
+ * @example
+ * ```ts
+ * import { isBuyFromListingSupported } from "thirdweb/extensions/marketplace";
+ *
+ * const supported = isBuyFromListingSupported(["0x..."]);
+ * ```
+ */
+export function isBuyFromListingSupported(availableSelectors: string[]) {
+  return BuyFromListing.isBuyFromListingSupported(availableSelectors);
```

Comment:
needs to check if `getListing` is supported and if `isListingValid` is supported if they call any extensions (which I think they do)?

Reply:
`isListingValid` seems to be clean. It's using isERC721 and isERC1155 to check the asset type but I dont think it's relevant in this context (?)